### ISSUE TITLE
Custom resource lambda like `custom-resource-apigw-cw-role` is not in vpc

### ISSUE THREAD

**Issue by zhaoyi0113** [serverless/serverless#11856]
### Are you certain it's a bug?

- [X] Yes, it looks like a bug

### Is the issue caused by a plugin?

- [X] It is not a plugin issue

### Are you using the latest v3 release?

- [X] Yes, I'm using the latest v3 release

### Is there an existing issue for this?

- [X] I have searched existing issues, it hasn't been reported yet

### Issue description

When an API gateway log is enabled, serverless will deploy a auto created lambda has the suffix `custom-resource-apigw-cw-role` in the name.  I have configured `provider -> vpc` in the serverless.ym file which attach VPC to all lambdas in this stack. But this lambda is missing from VPC. Is there a way to configure VPC on this lambda?

### Service configuration (serverless.yml) content

```yaml
NA
```

### Command name and used flags

serverless deploy

### Command output

```shell
command run successfully
```

### Environment information

```shell
not applicable
```

**Comment by medikoo** [comment]
@zhaoyi0113 I believe created custom resource lambda should be covered with VPC same way as other lambdas.

PR with a fix is welcome!

**Comment by Lokesh-Jawale** [comment]
Hi, @medikoo could you assign this issue to me? I would like to work on this.

**Comment by medikoo** [comment]
Sure @Lokesh-Jawale, assigned 👍

**Comment by patrickcollings** [comment]
We've just upgraded to 3.32 but this has caused us to have this error now when deploying, without any other changes. We've rolled back to the previous version in the mean time:
````
UPDATE_FAILED: CustomDashresourceDashapigwDashcwDashroleLambdaFunction (AWS::Lambda::Function)
Resource handler returned message: "The provided execution role does not have permissions to call CreateNetworkInterface on EC2
````

**Comment by medikoo** [comment]
@patrickcollings yes, this change was just reverted (see  #12001), and we're rolling out new release

**Comment by Lokesh-Jawale** [comment]
@medikoo I apologize for the inconvenience caused by the regression caused by my change.
if vpc is configured and CustomDashresourceDashapigwDashcwDashroleLambdaFunction lambda has to be in vpc, should I add the `AWSLambdaVPCAccessExecutionRole` policy for role `IamRoleCustomResourcesLambdaExecution`?

**Comment by medikoo** [comment]
@Lokesh-Jawale I'm not sure exactly, it appears that for many users with VPC it works as it's setup currently, and the patch broken they're deployments. We will need more feedback on it.
